Foundation/JSONSerialization.swift

```diff
@@ -475,6 +475,23 @@ private struct JSONWriter {
         switch num._cfTypeID {
         case CFBooleanGetTypeID():
             serializeBool(num.boolValue)
+        case CFNumberGetTypeID():
+            switch num._cfNumberType() {
+            case kCFNumberSInt8Type: fallthrough
+            case kCFNumberSInt16Type: fallthrough
+            case kCFNumberSInt32Type: fallthrough
+            case kCFNumberSInt64Type: fallthrough
+            case kCFNumberCharType: fallthrough
+            case kCFNumberShortType: fallthrough
+            case kCFNumberIntType: fallthrough
+            case kCFNumberLongType: fallthrough
+            case kCFNumberLongLongType:
```

Comment:
I don't think this is quite right... On a 32-bit platform `Int` is 32 bits, so some of these number types will overflow?